post_office/mail.py

```diff
@@ -171,22 +173,28 @@ def send_many(kwargs_list):
 
 def get_queued():
     """
-    Returns a list of emails that should be sent:
-     - Status is queued
-     - Has scheduled_time lower than the current time or None
+    Returns a list of emails that should be sent fulfilling these conditions:
+     - Status is queued or requeued
+     - Has scheduled_time before the current time or is None
+     - Has expires_at after the current time or is None
     """
-    return Email.objects.filter(status=STATUS.queued) \
-        .select_related('template') \
-        .filter(Q(scheduled_time__lte=now()) | Q(scheduled_time=None)) \
-        .order_by(*get_sending_order()).prefetch_related('attachments')[:get_batch_size()]
+    now = timezone.now()
+    query = (
+        (Q(status=STATUS.queued) | Q(status=STATUS.requeued)) &
+        (Q(scheduled_time__lte=now) | Q(scheduled_time__isnull=True)) &
```

Comment:
Nitpick: `Q(scheduled_time=None)`

Reply:
I believe that the proper way doing it in Django is using [...__isnull=True](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#isnull) otherwise their documentation would state that. The only mentioned alternative is [...__exact=None](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#exact), but then I really prefer `...__isnull=True`.

Reply:
I agree with the nitpick from @selwin.  I always use this approach and it is actually pretty readable.

Reply:
@Mogost , this might be true, but then I would like to know why it isn't mentioned anywhere in the documentation. From reading the docs, the `...__isnull=True`, seems to be the right approach.
Or I didn't read the docs properly. Do you have other sources?

Reply:
I only have django documentation. But I will try to convince you that the offer is good.
First of: 
> As a convenience when no lookup type is provided (like in Entry.objects.get(id=14)) the lookup type is assumed to be **exact**.

Second of:
> Exact match. If the value provided for comparison is None, it will be interpreted as an SQL NULL (see isnull for more details).

Therefore, what you mention as an alternative option can be finalized by taking into account the above.

Ultimately, this is just nitpicking.

Reply:
Yes, for datetime fields doing the query with `Q(scheduled_time=None)` gives exactly the same compiled SQL as `Q(scheduled_time__isnull=True)` does. But this is not true for all field types, for instance `JSONField`, where `fieldname=None` translates to Javascript Object Notation's `null`, where as `fieldname__isnull=True` translates to SQL's `IS NULL`.

Moreover, while `Q(scheduled_time=None)` may give the desired result, if one has to reverse the logic, `Q(scheduled_time!=None)` doesn't work for obvious reasons. So we have two way of spellings, `Q(scheduled_time=None)` to compile into `IS NULL` and `Q(scheduled_time__isnull=False)`  to compile into `IS NOT NULL`.

Don't you think that this makes things less explicit? Presumably that's the reason, why the official Django docs doesn't encourage its usage.